lib/puppet/util/windows/adsi.rb

```diff
@@ -94,6 +94,23 @@ def execquery(query)
       wmi_connection.execquery(query)
     end
 
+    # https://docs.microsoft.com/en-us/dotnet/api/microsoft.powershell.commands.domainrole?view=powershellsdk-1.1.0
+    DOMAIN_ROLES = {
+      STANDALONE_WORKSTATION: 0,
+      MEMBER_WORKSTATION: 1,
+      STANDALONE_SERVER: 2,
+      MEMBER_SERVER: 3,
+      BACKUP_DOMAIN_CONTROLLER: 4,
+      PRIMARY_DOMAIN_CONTROLLER: 5,
+    }
```

Comment:
does this need to be a hash? I see it more as multiple constants:

```ruby
      STANDALONE_WORKSTATION = 0
      MEMBER_WORKSTATION = 1
      STANDALONE_SERVER = 2
      MEMBER_SERVER = 3
      BACKUP_DOMAIN_CONTROLLER =4
      PRIMARY_DOMAIN_CONTROLLER= 5
      DOMAIN_ROLES = [STANDALONE_WORKSTATION, etc]
```

and then use it like `DOMAIN_ROLES.find?(query.DomainRole)`

Reply:
Also usually the convention is that constants are defined top level in the class